scipy/optimize/_basinhopping.py

```diff
@@ -396,12 +396,13 @@ def basinhopping(func, x0, niter=100, T=1.0, stepsize=0.5,
     niter_success : integer, optional
         Stop the run if the global minimum candidate remains the same for this
         number of iterations.
-    seed : int or `np.random.RandomState`, optional
-        If `seed` is not specified the `np.RandomState` singleton is used.
-        If `seed` is an int, a new `np.random.RandomState` instance is used,
-        seeded with seed.
-        If `seed` is already a `np.random.RandomState instance`, then that
-        `np.random.RandomState` instance is used.
+    seed : {int, `~np.random.RandomState`, `~np.random.Generator`}, optional
+        If `seed` is not specified the `~np.random.RandomState` singleton is
+        used.
+        If `seed` is an int, a new ``RandomState`` instance is used, seeded
```

Comment:
Should this default behavior be deprecated in favor of using the `np.random.Generator`? 

Reason I ask is because not sure how long `RandomState` is supported.

Reply:
`RandomState` is going to be supported forever, more or less.

https://numpy.org/neps/nep-0019-rng-policy.html#supporting-unit-tests

I do recommend moving towards using `Generator`, though, because it supports features that will be useful to these optimizers, and will be faster, in general. This current PR takes a good, measured approach to that move. I.e. _accept_ a `Generator` if one is provided, but otherwise continue to behave as before if provided a `seed`. That lets people slowly opt in. At some point, it might be wise to start creating `Generator`s from provided seeds, but that can be quite some time in the future.

Reply:
@rkern thanks for clarifying and passing along the link. Sounds like no deprecation warning need be emitted here.